lib/core-server/src/utils/watch-story-specifiers.ts

```diff
@@ -73,9 +73,14 @@ export function watchStorySpecifiers(
               // because the directoru could already be within the files part (e.g. './x/foo/bar')
               path.basename(specifier.files)
             );
-            const files = await glob(dirGlob);
+            // glob only supports forward slashes
+            const files = await glob(dirGlob.replace(/\\/g, '/'));
+
             files.forEach((filePath) => {
-              const fileImportPath = toImportPath(path.relative(options.workingDir, filePath));
+              const fileImportPath = toImportPath(
+                // use posix path separators even on windows
+                path.relative(options.workingDir, filePath).replace(/\\/g, '/')
+              );
```

Comment:
🚨  Non-test file.  The documentation for `glob` states that it requires forward slashes, not windows paths, so I think this might have been causing problems on windows, although I can't find any issues reporting it.